src/Nethermind/Nethermind.Blockchain.Test/TransactionSelectorTests.cs

```diff
@@ -221,13 +221,243 @@ public static IEnumerable EnoughShardBlobTransactionsSelectedTestCases
+
+        private static Transaction CreateBlobTransaction(Address address, PrivateKey key, UInt256 maxFee, int blobCount, UInt256 nonce, uint priority = 1)
+        {
+            return Build.A.Transaction.WithSenderAddress(address).WithType(TxType.Blob).WithNonce(nonce)
+                .WithMaxFeePerGas(maxFee)
+                .WithMaxFeePerBlobGas(2)
+                .WithMaxPriorityFeePerGas(priority)
+                .WithGasLimit(20)
+                .WithBlobVersionedHashes([.. Enumerable.Range(0, blobCount).Select(i => new byte[1] { 0 })])
+                .SignedAndResolved(key).TestObject;
+        }
+
+        public static IEnumerable BlobTransactionOrderingTestCases
+        {
+            get
+            {
+                (Address address, PrivateKey key)[] Accounts =
+                {
+                    (TestItem.AddressA, TestItem.PrivateKeyA),
+                    (TestItem.AddressB, TestItem.PrivateKeyB)
+                };
+
+                {
+                    var blobTxs = CreateTestCase();
+                    var txs = new List<Transaction>();
+
+                    UInt256 nonce = 1;
+                    AddTxs(txCount: 5, blobsPerTx: 5, account: 0, txs, ref nonce);
+                    AddTxs(txCount: 7, blobsPerTx: 1, account: 0, txs, ref nonce);
+
+                    blobTxs.Transactions = txs;
+
+                    blobTxs.ExpectedSelectedTransactions.AddRange(blobTxs.Transactions.Take(1));
+
+                    yield return new TestCaseData(blobTxs).SetName("Blob Transaction Ordering, Single Account, Single Blob");
+                }
+                {
+                    var blobTxs = CreateTestCase();
+                    var txs = new List<Transaction>();
+
+                    UInt256 nonce = 1;
+                    AddTxs(txCount: 5, blobsPerTx: 5, account: 0, txs, ref nonce);
+                    AddTxs(txCount: 1, blobsPerTx: 2, account: 0, txs, ref nonce);
+                    AddTxs(txCount: 5, blobsPerTx: 1, account: 0, txs, ref nonce);
+
+                    blobTxs.Transactions = txs;
+
+                    blobTxs.ExpectedSelectedTransactions.AddRange(blobTxs.Transactions.Take(1));
+
+                    yield return new TestCaseData(blobTxs).SetName("Blob Transaction Ordering, Single Account, Dual Blob");
+                }
+                {
+                    var blobTxs = CreateTestCase();
+                    var txs = new List<Transaction>();
+
+                    UInt256 nonce = 1;
+                    AddTxs(txCount: 5, blobsPerTx: 5, account: 0, txs, ref nonce);
+                    nonce = 1;
+                    AddTxs(txCount: 5, blobsPerTx: 1, account: 1, txs, ref nonce);
+
+                    blobTxs.Transactions = txs;
+
+                    blobTxs.ExpectedSelectedTransactions.AddRange(blobTxs.Transactions.Take(1));
+                    blobTxs.ExpectedSelectedTransactions.AddRange(blobTxs.Transactions.Skip(5).Take(1));
```

Comment:
Probably "2 txs with 6 blobs are better than 5 txs with 5 blobs" is not a typical case. Can you also add a case with bigger max fee (like we usually have big gas price and low blob gas price)

Reply:
AddTxs bumps priority fee so are higher with more blobs (`priority fee = blobs * 2`) if not specified, so 5 + 1 would be better in this case; have another test where priorty fees are all the same where it switches (and is what the hive test is checking and saying it should be 5+1 rather than 5 x 1; which as you point out isn't optimal)